CONTRIBUTING.md

```diff
@@ -101,3 +104,49 @@ The next section contains more information on the workflow followed for Pull Req
     - PRs can be closed by the author without merging
     - PRs may be closed by a Final Approver if the decision is made that the PR is not going to be merged 
 
+### Avoid the following mistakes!
+
+-  Fix a new issue and submit a PR without reporting and getting it approved at first.
+-   Fix an issue assigned to someone else and submit a PR before the assignee does.
+-   Report issues which are previously reported by others. (Please check the both open and closed issues before you report an issue).
+-   Suggest completely new features in the issue list. (Please use the mailing list/gitter channel for these kinds of suggestions. Use issue list to suggest bugs/features in the already implemented sections.)
+
+## Git & Github
+
+### Configure remotes
+
+When a repository is cloned, it has a default remote called `origin` that points to your fork on GitHub, not the original repository it was forked from. To keep track of the original repository, you should add another remote named `upstream`:<br />
+1. Get the path where you have your git repository on your machine. Go to that path in Terminal using cd. Alternatively, Right click on project in Github Desktop and hit ‘Open in Terminal’.<br />
+2. Run `git remote -v`  to check the status you should see something like the following:<br />
+> origin    https://github.com/YOUR_USERNAME/spec.git (fetch)<br />
+> origin    https://github.com/YOUR_USERNAME/spec.git (push)<br />
+3. Set the `upstream`:<br />
+ `git remote add upstream https://github.com/oam-dev/spec.git`<br />
+4. Run `git remote -v`  again to check the status, you should see something like the following:<br />
+> origin    https://github.com/YOUR_USERNAME/spec.git (fetch)<br />
+> origin    https://github.com/YOUR_USERNAME/spec.git (push)<br />
+> upstream https://github.com/oam-dev/spec.git  (fetch)<br />
+> upstream  https://github.com/oam-dev/spec.git (push)<br />
+5. To update your local copy with remote changes, run the following:<br />
+`git fetch upstream master`<br />
+ `git rebase  upstream/master`<br />
+This will give you an exact copy of the current remote, make sure you don't have any local changes.<br />
+6. Project set-up is complete.
+
+
+### Contributing and developing a feature
+
+1. Make sure you are in the develop branch `git checkout master`.<br />
+2. Sync your copy `git pull --rebase upstream master`.<br />
```

Comment:
This part is very kind to contributors, nice! But I think we can just say `Sync your copy with the upstream` instead of providing detailed synchronization method. There are two ways to do the sync, rebase and [merge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork). One can choose as they like.

Reply:
Ok